plugins/modules/instance.py

```diff
@@ -864,6 +881,96 @@ def _update_firewall(self) -> None:
                 "and 'firewall_device' modules."
             )
 
+    def _wait_for_instance_status(self, status: str) -> None:
+        def poll_func() -> bool:
+            self._instance.invalidate()
+            return self._instance.status == status
+
+        try:
+            poll_condition(poll_func, 4, self._timeout_ctx.seconds_remaining)
+        except polling.TimeoutException:
+            self.fail(
+                f"failed to wait for instance to reach status {status}: timeout period expired"
+            )
+
+    def _update_type(self) -> None:
+        """
+        Handles updates on the type field.
+        """
+
+        new_type = self.module.params.get("type")
+        auto_disk_resize = self.module.params.get("auto_disk_resize")
+        migration_type = self.module.params.get("migration_type")
+
+        previously_booted = self._instance.status == "running"
+
+        if new_type is None or new_type == self._instance.type.id:
```

Comment:
Should we also consider checking if `new_type` is a valid type value? Same for `new_region`?